src/OrbitGl/CaptureWindow.cpp

```diff
@@ -113,8 +113,21 @@ void CaptureWindow::PreRender() {
   }
 
   if (time_graph_ != nullptr) {
-    time_graph_->UpdateLayout();
-    UpdateChildrenPosAndSize();
+    const int kMaxLayoutLoops =
+        app_ != nullptr && (app_->IsCapturing() || app_->IsLoadingCapture()) ? 1 : 10;
+
+    int layout_loops = 0;
+
+    // Layout changes of one element may require other elements to be updated as well,
+    // so layouting needs to be done until all elements report that they do not need to
+    // be updated further. As layout requests bubble up, it's enough to check this for
+    // the root element (time graph) of the tree.
+    // During loading or capturing, only a single layouting loop is executed as we're
+    // streaming in data from a seperate thread.
+    do {
+      UpdateChildrenPosAndSize();
+      time_graph_->UpdateLayout();
+    } while (time_graph_->LayoutHasChanged() || ++layout_loops < kMaxLayoutLoops);
```

Comment:
Maybe log if we hit the maximum number of iterations? That would indicate a cyclic dependency in the UI elements which we should look into (?).

Reply:
There is a weird state during which we process data (finalizing capture), but app_ reports to no longer be streaming or loading the capture. During this time, I ran into the max layouting loops frequently, so the warning is a bit misleading. I wanted to create a bug for this and then add a warning, but I forgot, thanks for noticing this.